HW10/HW10_Interfaces_Abstract_classes/src/User.java

```diff
@@ -0,0 +1,104 @@
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.ArrayList;
+
+public class User implements UserInterface{
+
+    private String name;
+    private HashMap<String, User> contacts;
+    private ArrayList<Message> messages;
+
+
+    public User(String name) {
+        contacts = new HashMap<>(); // null pointer exeception
+        messages = new ArrayList<>();// null pointer exception
+        this.name = name;
+    }
+
+    // Додайте метод для читання повідомлення,
+    // який змінює його статус на "прочитане".
+    @Override
+    public String getUserName() {
+        return name;
+    }
+
+    @Override
+    public void addContact(String phone, User contact) {
+        contacts.put(phone, contact);
+        System.out.println("HashMap of contacts: " + contacts);
+        System.out.println("---");
+
+    }
+
+    public void deleteContact(String phone, User contact){
+        contacts.remove(phone, contact);
+        System.out.println("HashMap of contacts WHAT IS LEFT: " + contacts);
+        System.out.println("---");
+
+    }
+    @Override
+    public void getUserContacts() {
+        System.out.println("Hashmap of contacts" + contacts);
+        System.out.println("***");
+    }
+
+    @Override
+    public void getUserMessages() {
+        for (Message element : messages){
+           System.out.println("Message is the following " + element.toString());
+        }
+        //System.out.println(messages.toString()); і так, і так піде
+        System.out.println("---");
+    }
+
+
+    @Override
+    public void sendMessage(Message message, User receiver) {
+        message.setMessageStatus(MessageStatuses.SENT);
```

Comment:
it makes sense to change the status of the message **after** the rest of operations withing the method are done
In this case if any error occurs before the status change - status won't be set